build_tools/teatime.py

```diff
@@ -178,6 +292,9 @@ def main(cl_args: list[str]):
         default=False,
         help="Log timestamps along with log lines to the log file",
     )
+    p.add_argument(
+        "--skip-index", action="store_true", help="Skip indexing and uploading logs"
+    )
```

Comment:
I don't see value in having this argument in teatime.py. It's a big mode switch to have `cmake/therock_subproject.cmake` run one command and then `.github/workflows/build_*_packages.yml` run another command. At that point just have the workflow files run the index generator and upload script directly.

If keeping it, flipping the condition from opt-out to opt-in would make more sense to me. That way, running the script on a single file would not generate an index page.